src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/adyen/paymentFromComponent.js

```diff
@@ -11,25 +11,140 @@ const collections = require('*/cartridge/scripts/util/collections');
 const AdyenHelper = require('*/cartridge/scripts/util/adyenHelper');
 const AdyenLogs = require('*/cartridge/scripts/adyenCustomLogs');
 
+const expressMethods = ['applepay', 'amazonpay'];
+
+function setBillingAndShippingAddress(reqDataObj, currentBasket) {
+  const shopperDetails = reqDataObj.customer;
+
+  let { billingAddress } = currentBasket;
+  Transaction.wrap(() => {
+    if (!billingAddress) {
+      billingAddress = currentBasket.createBillingAddress();
+    }
+    billingAddress.setFirstName(shopperDetails.billingAddressDetails.firstName);
+    billingAddress.setLastName(shopperDetails.billingAddressDetails.lastName);
+    billingAddress.setPhone(shopperDetails.profile.phone);
+    billingAddress.setAddress1(shopperDetails.billingAddressDetails.address1);
+    billingAddress.setCity(shopperDetails.billingAddressDetails.city);
```

Comment:
Good point! Will definitely add the state/county.
Basically address1 is the mandatory field that comes from apple configuration, containing the street, and address2 is the optional one, which we are setting only in case its populated.